src/nvim/liveupdate.c

```diff
@@ -0,0 +1,195 @@
+#include "nvim/liveupdate.h"
+#include "nvim/memline.h"
+#include "nvim/api/private/helpers.h"
+#include "nvim/msgpack_rpc/channel.h"
+
+// Register a channel. Return True if the channel was added, or already added.
+// Return False if the channel couldn't be added because the buffer is
+// unloaded.
+bool liveupdate_register(buf_T *buf, uint64_t channel_id)
+{
+  // must fail if the buffer isn't loaded
+  if (buf->b_ml.ml_mfp == NULL) {
+    return false;
+  }
+
+  // count how many channels are currently watching the buffer
+  size_t size = kv_size(buf->liveupdate_channels);
+  if (size) {
```

Comment:
This `if` looks redundant, as the for loop will already not run if `size == 0`.

Reply:
Good catch! Thanks for taking the time to review the changes BTW.